### PR TITLE
Changed random to roll, added dice for dnd steez

### PR DESCRIPTION
**Pull request by donk**
Changed it up a bit to have a default roll using a very cool number. I like having no args on a roll because it lets you quickly shoot one off with minimal typing. 

The downside is that we'd need to have a 'help' command that will give a description of commands so they know how to use it, but I like that idea anyway; gives us more ability to describe commands.

### CODE REVIEW

File: client/src/commands/index.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import { sendEvent } from "../network"
 import { CHAT_EVENT, INPUT_EVENT } from "../../../events"
 import { commandEmitter } from "./emitter"
-import { Random } from "./random"
+import Roll from './roll';
 
 const commandModules = [
-  Random,
+  Roll
```
Comment (Andr3wRiv3rs) [inline client/src/commands/index.ts:7]: Needs a trailing comma, helps with merge conflicts.

File: client/src/commands/roll.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import { LOG_EVENT, ROLL_EVENT } from "../../../events"
+import { LogItem } from "../components/terminal"
+import { sendEvent } from "../network"
+import { commandEmitter, CommandModule } from "./emitter"
+
+export const diceRegex =/(\d+)d(\d+)/;
+
+export type DiceProps = {
```
Comment (Andr3wRiv3rs) [inline client/src/commands/roll.ts:8]: `interface` is more appropriate than `type` here.

File: client/src/commands/roll.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  count: number;
```
Comment (Andr3wRiv3rs) [inline client/src/commands/roll.ts:9]: Don't need semicolons anywhere, but doesn't hurt anything either.

File: client/src/commands/roll.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+const defaultRoll: DiceProps = {
+  count: 1,
+  sides: 69,
+}
+
+const parseDice = (value: string) => {
+  const match = value.match(diceRegex);
+  if (!match) return false;
+  if (match.length < 3) return false;
+  return {count: parseInt(match[1]), sides: parseInt(match[2])};
```
Comment (Andr3wRiv3rs) [inline client/src/commands/roll.ts:22]: this should be split onto multiple lines to avoid conflicts

File: client/src/commands/roll.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+const parseNumber = (value: string) => {
+  let result: DiceProps | boolean;
+  
+  if (!value) return defaultRoll;
+
+  if (result = parseDice(value)) return result;
+
+  if (!isNaN(parseInt(value))){
+    return {count: 1, sides: parseInt(value)};
```
Comment (Andr3wRiv3rs) [inline client/src/commands/roll.ts:33]: see last comment

File: server/network/events/roll.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+const rollDice = (dice: DiceProps) => {
+  let total: number = 0;
+  for (let i = 0; i < dice.count; i++){
+    total += Math.floor(Math.random() * dice.sides) + 1;
+  }
+  return total;
+}
+
+const handler: NetworkEventHandler = (socket, dice: DiceProps, player) => {
```
Comment (Andr3wRiv3rs) [inline server/network/events/roll.ts:18]: you can do `player: Player` here

**Review by Andr3wRiv3rs — CHANGES_REQUESTED** [review]

File: client/src/commands/roll.ts (hunk 4 above)
Comment (donk) [reply client/src/commands/roll.ts:22]: This makes so much sense and I've never thought about it

**Pull request merged**